### PR TITLE
adding_unsubscribe

### PR DESCRIPTION
**Pull request by nbsven**

### CODE REVIEW

File: src/main/java/com/epam/magazinestore/controller/AccountController.java [hunk 1]
```diff
@@ -65,7 +71,7 @@ public ModelAndView account(@RequestParam("name") String name, @RequestParam("pa
     @GetMapping(value = LOGIN)
     public ModelAndView login(ModelAndView modelAndView, HttpSession session) {
         String message = (String) session.getAttribute("message");
-        if(message != null) {
+        if (message != null) {
             modelAndView.addObject("message", message);
             session.removeAttribute("message");
         }
```
Comment (Jebsuz) [inline src/main/java/com/epam/magazinestore/controller/AccountController.java]: what is this? what does it do?

File: src/main/java/com/epam/magazinestore/controller/AccountController.java [hunk 2]
```diff
@@ -111,4 +117,12 @@ public ModelAndView user(HttpSession session) {
         return mav;
     }
 
+    @PostMapping(value = "/unsubscribe")
+    public String unsubscribe(@RequestParam(name = "magazineId", defaultValue = "1") long magazineId,
+                              HttpSession session) {
+        Account account = (Account) session.getAttribute(SESSION_ACCOUNT);
+        subscriptionService.unsubscribe(accountService.getAccountByUsername(account.getUsername()), magazineService.getMagazineById(magazineId));
```
Comment (Jebsuz) [inline src/main/java/com/epam/magazinestore/controller/AccountController.java]: Such saving in lines of codes leads to bad readablitiy, if you would explicitly extract `String userName = account.getUsername()` and then explicitly get `Account userAccount = accountService.getAccountByUsername(userName)` and the result would be passed as first argument of `subscriptionService.unsubscribe(userAccount, ...)`.
When you inline such things code, you forcing the reader to return backwards each time he evalutes method parameter to remind him self why and what is happening. Maintainability is one of the keys to success

File: src/main/java/com/epam/magazinestore/controller/AccountController.java [hunk 3]
```diff
@@ -65,7 +71,7 @@ public ModelAndView account(@RequestParam("name") String name, @RequestParam("pa
     @GetMapping(value = LOGIN)
     public ModelAndView login(ModelAndView modelAndView, HttpSession session) {
         String message = (String) session.getAttribute("message");
-        if(message != null) {
+        if (message != null) {
```
Comment (AntonKiselyov) [inline src/main/java/com/epam/magazinestore/controller/AccountController.java]: Objects.nonNull()

File: src/main/java/com/epam/magazinestore/controller/AccountController.java [hunk 4]
```diff
@@ -65,7 +71,7 @@ public ModelAndView account(@RequestParam("name") String name, @RequestParam("pa
...
             modelAndView.addObject("message", message);
```
Comment (AntonKiselyov) [inline src/main/java/com/epam/magazinestore/controller/AccountController.java]: Message is a constant

File: src/main/java/com/epam/magazinestore/controller/AccountController.java [hunk 5]
```diff
@@ -100,7 +106,7 @@ public ModelAndView createAccount(@RequestParam("name") String name, @RequestPar
     public ModelAndView user(HttpSession session) {
         Account account = (Account) session.getAttribute(SESSION_ACCOUNT);
         ModelAndView mav = new ModelAndView();
-        if(account != null) {
+        if (account != null) {
             mav.addObject(NAME, account.getUsername());
             mav.setViewName(USER_JSP);
             List<Subscription> subscriptions = subscriptionService.getSubscriptionsByAccount(account, 0, 10);
```
Comment (AntonKiselyov) [inline src/main/java/com/epam/magazinestore/controller/AccountController.java]: What are 0 and 10 ?

File: src/main/webapp/WEB-INF/jsp/user-page.jsp [hunk 6]
```diff
@@ -4,41 +4,47 @@
...
-
-        <form action="../sign_out">
-            <input value="end", type="submit">
-        </form>
-
-    </body>
+<body>
+<br>
+
+<p>Your name is ${account_name}</p> <br>
```
Comment (AntonKiselyov) [inline src/main/webapp/WEB-INF/jsp/user-page.jsp]: I guess this strings are titles and you should placed it in .properties

**Pull request merged**